### PR TITLE
Ports wounds from TG

### PR DESCRIPTION
**Pull request by SomeguyManperson**
port list (will be updated as I get more stuff done):
tgstation/tgstation#50558
tgstation/tgstation#52004
tgstation/tgstation#51806
tgstation/tgstation#51850
tgstation/tgstation#51786
tgstation/tgstation#51845
tgstation/tgstation#52193
tgstation/tgstation#53973
tgstation/tgstation#52534
tgstation/tgstation#53374
tgstation/tgstation#54922
tgstation/tgstation#54546
tgstation/tgstation#53349
tgstation/tgstation#53763
tgstation/tgstation#52688
tgstation/tgstation#56056
tgstation/tgstation#52995
tgstation/tgstation#52951
tgstation/tgstation#53695
tgstation/tgstation#59848
tgstation/tgstation#59865

# Github documenting your Pull Request

ports tg's wound system, which adds negative effects on taking high damage without protection that have to be treated seperately from normal damage.
pring this as a draft currently so I can check diffs easier and so people can downvote 200 times because it makes the game harder, currently it compiles and functions mostly as intended

# Wiki Documentation

good luck lmao

# Changelog

:cl:  theos for porting, ryll ryll for the whole wounds system and a bunch of tg coders for assorted fixes and balancing
rscadd: ports wounds from tg, I said I'd do this like a few months ago and was not lying
bugfix: holdup fixed to not cause pulling the target to fire turns out my old method was stupid and didn't work
/:cl:

### CODE REVIEW

**Comment by adamsong** [comment]
> code works lmao

> Some checks were not successful

**Comment by SomeguyManperson** [comment]
merge conflict, remember to pull from master  before going on a year long attempt to port a single feature guys

**Comment by Cartlord** [comment]
But for what reason?
This place is meant to be new-player friendly, and I don't see how the wounds system is new-player friendly.

**Comment by SomeguyManperson** [comment]
> But for what reason?
> This place is meant to be new-player friendly, and I don't see how the wounds system is new-player friendly.

making the game extremely surface level and shallow isn't new player friendly it's boring, if you want a boring game go play roblox rp

**Comment by Marmio64** [comment]
med system is so lame and boring right now, there is no difference between field care and medbay care since practically every injury can be fixed on the go very easily and without outside help. Plus this could maybe open up additional ways to use the rudimentary skill system we have.

**Comment by maxion12345** [comment]
haven't messed around with wounds, no clue how they work

**Comment by Mqiib** [comment]
Very cool, thank you ultra excellent cooder theos man

**Comment by Sniblet** [comment]
Doesn't TG hate this
Doesn't it nerf antags hard because the only cure is surgery which only a few of them have time/friends for

**Comment by SomeguyManperson** [comment]
every wound type has a ghetto alternative to proper surgery/medical treatment if it needs that, it might be a bit difficult or have some downside but it'll work

**Comment by Marmio64** [comment]
just give as many antags as possible a way to heal the wounds easier, like changeling fleshmend and revive healing them, ninja suit could auto heal them, slings and darkspawn autoheal, cheap traitor item for healing wounds, stuff like that if the concern is nerfed antags.

**Comment by Sniblet** [comment]
Every monkey on the map infinitely bleeds now, it's probably this doing it

**Comment by maxion12345** [comment]
honestly wounds sound cool but having to do surgery to fix somethings that just aren't critical damage is a bit cringe

**Comment by SomeguyManperson** [comment]
Cuts: generally heal themselves over time unless they are critical, heal faster with application of cauterizing heat from hot stuff or laser guns or gauze
Pierces: like cuts but they don't heal over time unless your body temperature is slightly cold and bleed less
Burns: heal from application of basic anti-burn medicine or a paramedic's penlight
Bones: basic level can be manually re-inserted by crushing yourself with an airlock or a bonesetter, higher levels require the application of bone paste (which can be done manually but has a failure chance reduced by painkillers) but can be splinted to reduce the negative effects (it's supposed to disable the limb at higher severity without a splint but doesnt currently)
Surgery is generally an option that works but it's never required if you really can't use it, the only thing that really requires proper medical equipment is bone breakage which won't kill you eventually just by existing

Also shadowlings and darkspawn SHOULD be immune to wounding, the only antags that can't handle the damage well are currently ninjas and clock cult and I'm not sure how I want clock cult to handle it since making sentinel's delete them seems a bit cheap

**Comment by Marmio64** [comment]
> Cuts: generally heal themselves over time unless they are critical, heal faster with application of cauterizing heat from hot stuff or laser guns or gauze
> Pierces: like cuts but they don't heal over time unless your body temperature is slightly cold and bleed less
> Burns: heal from application of basic anti-burn medicine or a paramedic's penlight
> Bones: basic level can be manually re-inserted by crushing yourself with an airlock or a bonesetter, higher levels require the application of bone paste (which can be done manually but has a failure chance reduced by painkillers) but can be splinted to reduce the negative effects (it's supposed to disable the limb at higher severity without a splint but doesnt currently)
> Surgery is generally an option that works but it's never required if you really can't use it, the only thing that really requires proper medical equipment is bone breakage which won't kill you eventually just by existing
> 
> Also shadowlings and darkspawn SHOULD be immune to wounding, the only antags that can't handle the damage well are currently ninjas and clock cult and I'm not sure how I want clock cult to handle it since making sentinel's delete them seems a bit cheap

Give ninja suit a button that lets it heal a single wound with a decent cooldown perhaps, and sentinel's compromise could tend one wound per use but converts it to a bunch of toxin damage.

**Comment by nmajask** [comment]
Probably should add a way for ghosts to see wounds, like how they can with health and chems.

**Comment by SomeguyManperson** [comment]
examining someone will show all wounds, the only thing the med scanner does is show treatment process and the specific name so I don't think it matters nearly as much

**Comment by burn1cycle** [comment]
Absolutely not. I've fought for years to keep this out because this is baystation and baystation already exists and it's also unpopular due to the tediousness of it's systems. We also don't have enough competent players to support this feature.

**Comment by SomeguyManperson** [comment]
Wounds aren't tedious, if you took even a minute to find out how they worked you'd know this but you didn't, you just hate it because it's new. If you have an argument against it go right ahead but you don't, it's just "I don't like it because it's bad, I don't have to do research because I know I'm right"

**Comment by burn1cycle** [comment]
> Wounds aren't tedious, if you took even a minute to find out how they worked you'd know this but you didn't, you just hate it because it's new. If you have an argument against it go right ahead but you don't, it's just "I don't like it because it's bad, I don't have to do research because I know I'm right"

I've played Baystation and I know about the ridiculous wound system. You get fucked up from a minor injury and then you're crippled until medbay fixes you. I'm not having it.

 I'm going to tell you for fucking fact this is ruining our game. You have ruined the game. Not only is getting injured more shitty but if you injure others it suddenly becomes a bigger deal in AHelps because it reduces playable time from others as well. I swear to fucking god if an admin hits me or anyone with "OMG the wounds you caused!" we should all blame it on you.

 I guess it's been nice having you here but since you're leaving us for Baystation it's farewall. Later dude.

**Comment by LoliconSlayer** [comment]
> > Wounds aren't tedious, if you took even a minute to find out how they worked you'd know this but you didn't, you just hate it because it's new. If you have an argument against it go right ahead but you don't, it's just "I don't like it because it's bad, I don't have to do research because I know I'm right"
> 
> I've played Baystation and I know about the ridiculous wound system. You get fucked up from a minor injury and then you're crippled until medbay fixes you. I'm not having it.
> 
>  I'm going to tell you for fucking fact this is ruining our game. You have ruined the game. Not only is getting injured more shitty but if you injure others it suddenly becomes a bigger deal in AHelps because it reduces playable time from others as well. I swear to fucking god if an admin hits me or anyone with "OMG the wounds you caused!" we should all blame it on you.
> 
>  I guess it's been nice having you here but since you're leaving us for Baystation it's farewall. Later dude.

Theos ain't going anywhere just because you cried about his PR

**Review by burn1cycle — CHANGES_REQUESTED** [review]
revert

**Review by burn1cycle — CHANGES_REQUESTED** [review]
revert

**Comment by maxion12345** [comment]
I'ma be honest after fucking around with wounds for a bit they seem a bit inconsistent sometimes (like stuff happens that you would think gets you a wound, but it didn't). So far, so good! Although I wish puncture wound side was decided by bullet size, because I shot someone with a mosin and they only bled a little bit :(

But it's also very straightforward to treat.  There is even a dedicated medical item that gives step by step details to fixing wounds for people unfamiliar. Can we make it so I can't kill myself with the first aid scanner please though?

**Comment by SomeguyManperson** [comment]
> I wish puncture wound side was decided by bullet size, because I shot someone with a mosin and they only bled a little bit :(

The mosin has a wounding modifier of -35 probably since it already deals a massive amount of damage on its own, it still has a high chance to cause wounds on each hit
this effectively means its default roll of 1-145 has to be 65 to cause a moderate wound (default 30) 85 to cause a severe wound (default 50) and 135 (default 100) to cause a critical one, assuming no other present armor or effects

**Comment by SomeguyManperson** [comment]
> > Wounds aren't tedious, if you took even a minute to find out how they worked you'd know this but you didn't, you just hate it because it's new. If you have an argument against it go right ahead but you don't, it's just "I don't like it because it's bad, I don't have to do research because I know I'm right"
> 
> I've played Baystation and I know about the ridiculous wound system. You get fucked up from a minor injury and then you're crippled until medbay fixes you. I'm not having it.
> 
> I'm going to tell you for fucking fact this is ruining our game. You have ruined the game. Not only is getting injured more shitty but if you injure others it suddenly becomes a bigger deal in AHelps because it reduces playable time from others as well. I swear to fucking god if an admin hits me or anyone with "OMG the wounds you caused!" we should all blame it on you.
> 
> I guess it's been nice having you here but since you're leaving us for Baystation it's farewall. Later dude.

lol you still have no clue how this works don't you

**Comment by LoliconSlayer** [comment]
> revert


https://user-images.githubusercontent.com/54045237/129503813-7fab6b45-326d-40f6-839a-1b1164b001f8.mp4

**Comment by Sniblet** [comment]
Recently played with an autophagocytosis necrosis virus and the victims didn't bleed at all
Between the heparin and the huge amount of brute damage you would think that they would a lot, and they used to, but they didn't. I don't want to live in a world where A.N. doesn't turn humans into blood waterfalls anymore

File: code/datums/wounds/slash.dm [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+		return FALSE
+
+	if(!iscatperson(user))
+		return FALSE
+
+	lick_wounds(user)
+	return TRUE
+
+/// if a felinid is licking this cut to reduce bleeding
+/datum/wound/slash/proc/lick_wounds(mob/living/carbon/human/user)
```
Comment (SomeguyManperson) [inline code/datums/wounds/slash.dm]: this is kind of cringe should I keep it

**Comment by adamsong** [comment]
CPR should have a chance to do wounds to the chest

**Comment by Cartlord** [comment]
did you forget about mourning poultices or are they just not supposed to help with wounds (did a quick ctrl-f for "poultice", got nothing)

**Review by JamieD1 — APPROVED** [review]

**Comment by burn1cycle** [comment]
Can the wound system have some mercy for getting smacked around in monstermos? I was in a round where the AI changed pressures (as antag) to antag us and just getting blown into walls and stuff was enough to cause wounds. 

 I'm not even sure if TG has this brutal atmos system that blows you around, hey I could be wrong don't crucify me but I don't think they have it.

**Pull request merged**